dataactbroker/handlers/fileHandler.py

```diff
@@ -389,3 +411,144 @@ def getRss(self):
+
+        if not (StringCleaner.isNumeric(submission_id) and StringCleaner.isDate(start_date) and StringCleaner.isDate(end_date)):
+            exc = ResponseException("submission id, start, and/or end cannot be parsed into their appropriate types", StatusCode.CLIENT_ERROR)
+            return JsonResponse.create(exc, exc.status)
+
+        cgac_code = self.jobManager.getSubmissionById(submission_id).cgac_code
+        get_url = CONFIG_BROKER["d1_url"].format(cgac_code, start_date, end_date)
+
+        jq = JobQueue(job_queue_url=CONFIG_JOB_QUEUE['url'])
+
+        # Generate and upload D1 file to S3
+        user_id = LoginSession.getName(session)
+        timestamped_name = s3UrlHandler.getTimestampedFilename(CONFIG_BROKER["d1_file_name"])
+        upload_file_name = "".join([str(user_id), "/", timestamped_name])
+        d_file_id = self.jobManager.createDFileMeta(submission_id, start_date, end_date, "d1", CONFIG_BROKER["d1_file_name"], upload_file_name)
+        self.jobManager.setDFileStatus(d_file_id, "waiting")
+        jq.generate_d_file.delay(get_url, CONFIG_BROKER["d1_file_name"], user_id, d_file_id, InterfaceHolder, timestamped_name, skip_gen=True)
+
+        # Check status for D1 file
+        return self.checkD1File()
+
+    def checkD1File(self, submission_id=None):
+        """ Check status of D1 file generation """
+        if submission_id is None:
+            requestDict = RequestDictionary(self.request)
+            if not (requestDict.exists("submission_id")):
+                exc = ResponseException("Check D1 Files route requires submission_id", StatusCode.CLIENT_ERROR)
+                return JsonResponse.create(exc, exc.status)
+
+            submission_id = requestDict.getValue("submission_id")
+            if not (StringCleaner.isNumeric(submission_id)):
+                exc = ResponseException("submission id cannot be parsed into its appropriate type", StatusCode.CLIENT_ERROR)
+                return JsonResponse.create(exc, exc.status)
+
+        d1_file = self.jobManager.getDFileForSubmission(submission_id, "d1")
+        if d1_file is None:
+            response = {"status": "invalid", "url": "", "start": "", "end": "", "message": ""}
+            return JsonResponse.create(StatusCode.OK, response)
+
+        status = self.jobManager.getJobStatusNameById(d1_file.status_id)
+
+        if status == "finished" and not d1_file.is_submitted and False: # TODO: Remove "and False" when submission of D1 is desired
+            job_id = self.jobManager.getJobBySubmissionFileTypeAndJobType(submission_id, "award_procurement", "file_upload").job_id
+            self.jobManager.updateDFileName(submission_id, d1_file.upload_file_name, "award")
+            result = self.finalize(jobId=job_id)
+            self.jobManager.markDFileAsSubmitted(d1_file.d_file_id)
+            if result.status_code != 200:
+                raise ResponseException(result.data)
+
+        url = "" if d1_file.url is None else d1_file.url
```

Comment:
Yes it is. It stores the download link. I think the expiration was like 14k seconds, can't remember where I saw that though but yes, that should be handled somehow. If they come back a month later, they won't be able to download those files again with that link